### PR TITLE
examples: add RNN examples

### PR DESCRIPTION
**Pull request by shu1chen**
# Description

Add examples for Vanilla RNN and Linear-before-reset GRU RNN by referring to the [RNN descriptions in oneDNN documentation](https://oneapi-src.github.io/oneDNN/dev_guide_rnn.html).
For lbr_gru, the gates order is update, reset and output gate, and there is an extra bias for u' gate.

# Checklist

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [x] Have you formatted the code using clang-format?

### CODE REVIEW

**Review by mgouicem — APPROVED** [review]
Thank you very much for the PR.
Could you please change the commit names to align with oneDNN contribution guidelines?
Something like:
`examples: add vanilla_rnn example`
`examples: add lbr_gru example`

Could you also please add reference to those examples in the documentation? (at the end of file doc/primitives/rnn.md)

Thanks you!

File: examples/primitives/vanilla.cpp [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    write_to_dnnl_memory(bias_data.data(), bias_mem);
+    write_to_dnnl_memory(weights_layer_data.data(), user_weights_layer_mem);
+    write_to_dnnl_memory(weights_iter_data.data(), user_weights_iter_mem);
+
+    // Create memory descriptors for weights with format_tag::any. This enables
+    // the Vanilla primitive to choose the optimized memory layout.
+    auto vanilla_weights_layer_md
+            = memory::desc(weights_dims, dt::f32, tag::any);
+    auto vanilla_weights_iter_md
+            = memory::desc(weights_dims, dt::f32, tag::any);
```
[the remark this answers is not in the document]
Comment (shu1chen) [reply examples/primitives/vanilla.cpp]: Yes, it makes sense. Updated

File: examples/primitives/vanilla.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+/// [Vanilla RNN](@ref dev_guide_rnn) primitive in forward inference propagation
+/// mode.
+///
+/// Key optimizations included in this example:
+/// - Creation of optimized memory format from the primitive descriptor.
+///
+/// @page vanilla_example_cpp Vanilla RNN Primitive Example
+/// @copydetails vanilla_example_cpp_short
+///
+/// @include vanilla.cpp
```
[the remark this answers is not in the document]
Comment (shu1chen) [reply examples/primitives/vanilla.cpp]: Yes, updated

File: examples/primitives/lbr_gru.cpp [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    dnnl::engine engine(engine_kind, 0);
+
+    // Create dnnl::stream.
+    dnnl::stream engine_stream(engine);
+
+    // Tensor dimensions.
+    const memory::dim N = 2, // batch size
+            T = 3, // time steps
+            C1 = 2, // src channels
+            C2 = 3, // dst channels
```
[the remark this answers is not in the document]
Comment (shu1chen) [reply examples/primitives/lbr_gru.cpp]: Yes, updated

**Comment by shu1chen** [comment]
Thanks for reviewing. I have updated the commit messages and also added reference to those examples in the documentation.

**Review by mgouicem — APPROVED** [review]
Thanks for the PR!

**Pull request merged**